Source/WebCore/Modules/highlight/Highlight.h

```diff
@@ -32,42 +32,60 @@
 
 namespace WebCore {
 
+class AbstractRange;
 class CSSStyleDeclaration;
 class DOMSetAdapter;
 class PropertySetCSSStyleDeclaration;
-class StaticRange;
 
-
-struct HighlightRangeData : RefCounted<HighlightRangeData>, public CanMakeWeakPtr<HighlightRangeData> {
-    HighlightRangeData(Ref<StaticRange>&& range)
-        : range(WTFMove(range))
+class HighlightRangeData : public RefCounted<HighlightRangeData>, public CanMakeWeakPtr<HighlightRangeData> {
+public:
+    static Ref<HighlightRangeData> create(Ref<AbstractRange>&& range)
     {
+        return adoptRef(*new HighlightRangeData(WTFMove(range)));
     }
-    static Ref<HighlightRangeData> create(Ref<StaticRange>&& range)
+
+    AbstractRange& range() const { return m_range.get(); }
+    const std::optional<Position>& startPosition() const { return m_startPosition; }
+    void setStartPosition(Position&& startPosition) { m_startPosition = WTFMove(startPosition); }
+    const std::optional<Position>& endPosition() const { return m_endPosition; }
+    void setEndPosition(Position&& endPosition) { m_endPosition = WTFMove(endPosition); }
+
+private:
+    explicit HighlightRangeData(Ref<AbstractRange>&& range)
+        : m_range(WTFMove(range))
     {
-        return adoptRef(*new HighlightRangeData(WTFMove(range)));
     }
-    Ref<StaticRange> range;
-    std::optional<Position> startPosition;
-    std::optional<Position> endPosition;
+
+    Ref<AbstractRange> m_range;
+    std::optional<Position> m_startPosition;
```

Comment:
Since `Position` has a null value, surprised we need to use `std:.optional`. Also wondering why we don’t use the simpler `BoundaryPoint` type.